# Loan Status Prediction
This file contains project: Loan Status Prediction Using Machin Learning

## Problem Statement
Company wants to automate the loan eligibility process (real time) based on customer detail provided while filling online application form. These details are Gender, Marital Status, Education, Number of Dependents, Income, Loan Amount, Credit History and others. To automate this process, they have given a problem to identify the customers segments, those are eligible for loan amount so that they can specifically target these customers.

## About Dataset
The Data is taken from Kaggle(https://www.kaggle.com/datasets/ninzaami/loan-predication)

The format of the data:

- Variable: Description
- Loan_ID: Unique Loan ID
- Gender: Male/ Female
- Married: Applicant married (Y/N)
- Dependents: Number of dependents
- Education: Applicant Education (Graduate/ Under Graduate)
- Self_Employed: Self employed (Y/N)
- ApplicantIncome: Applicant income
- CoapplicantIncome: Coapplicant income
- LoanAmount: Loan amount in thousands
- Loan_Amount_Term: Term of loan in months
- Credit_History: Credit history meets guidelines
- Property_Area: Urban/ Semi Urban/ Rural
- Loan_Status: Loan approved (Y/N)

## Importing Libraries and dependencies

In [1]:
# Common Libs
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb

%matplotlib inline

In [ ]:
# Important function imports

In [ ]:
# model imports

## Data Collection

In [2]:
loan_data = pd.read_csv("Loan_data.csv")

In [3]:
loan_data.sample(5)

Loan_ID Gender Married Dependents     Education Self_Employed  \
132  LP001270   Male     Yes         3+  Not Graduate           Yes   
572  LP002168   Male      No          0      Graduate            No   
537  LP002107   Male     Yes         3+  Not Graduate            No   
35   LP001067   Male      No          0  Not Graduate            No   
303  LP001636   Male     Yes          0      Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
132             8000              250.0       187.0             360.0   
572             5333             2400.0       200.0             360.0   
537             2855              542.0        90.0             360.0   
35              2400             2400.0       123.0             360.0   
303             4600                0.0        73.0             180.0   

     Credit_History Property_Area Loan_Status  
132             1.0     Semiurban           N  
572             0.0         Rural           N  
537             1.0         Urban           N  
35              1.0     Semiurban           N  
303             1.0     Semiurban           Y

Data is loaded correctly

In [4]:
list(loan_data.columns)

['Loan_ID',
 'Gender',
 'Married',
 'Dependents',
 'Education',
 'Self_Employed',
 'ApplicantIncome',
 'CoapplicantIncome',
 'LoanAmount',
 'Loan_Amount_Term',
 'Credit_History',
 'Property_Area',
 'Loan_Status']

In [5]:
loan_data.shape

(981, 13)

total 981 different data points available

## Data Analysis and feature Engneering

First see type of data stored 

In [6]:
loan_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 981 entries, 0 to 980
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            981 non-null    object 
 1   Gender             957 non-null    object 
 2   Married            978 non-null    object 
 3   Dependents         956 non-null    object 
 4   Education          981 non-null    object 
 5   Self_Employed      926 non-null    object 
 6   ApplicantIncome    981 non-null    int64  
 7   CoapplicantIncome  981 non-null    float64
 8   LoanAmount         954 non-null    float64
 9   Loan_Amount_Term   961 non-null    float64
 10  Credit_History     902 non-null    float64
 11  Property_Area      981 non-null    object 
 12  Loan_Status        981 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 99.8+ KB


As loan ID is not important to us for now

In [7]:
loan_data.drop(columns=["Loan_ID"], inplace=True)

In [8]:
loan_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 981 entries, 0 to 980
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Gender             957 non-null    object 
 1   Married            978 non-null    object 
 2   Dependents         956 non-null    object 
 3   Education          981 non-null    object 
 4   Self_Employed      926 non-null    object 
 5   ApplicantIncome    981 non-null    int64  
 6   CoapplicantIncome  981 non-null    float64
 7   LoanAmount         954 non-null    float64
 8   Loan_Amount_Term   961 non-null    float64
 9   Credit_History     902 non-null    float64
 10  Property_Area      981 non-null    object 
 11  Loan_Status        981 non-null    object 
dtypes: float64(4), int64(1), object(7)
memory usage: 92.1+ KB


Checking for duplicates

In [9]:
loan_data.duplicated().sum()

2

2 duplicates doesn't make that much diffrence can be ignored

Checking for Missing values

In [10]:
loan_data.isnull().sum()

Gender               24
Married               3
Dependents           25
Education             0
Self_Employed        55
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount           27
Loan_Amount_Term     20
Credit_History       79
Property_Area         0
Loan_Status           0
dtype: int64

In [11]:
#droping missing values
loan_data = loan_data.dropna()

In [12]:
loan_data.isnull().sum()

Gender               0
Married              0
Dependents           0
Education            0
Self_Employed        0
ApplicantIncome      0
CoapplicantIncome    0
LoanAmount           0
Loan_Amount_Term     0
Credit_History       0
Property_Area        0
Loan_Status          0
dtype: int64

In [13]:
loan_data.shape

(769, 12)

## Seprarting Data and Label

## Train Test Split

## Model Selection, Training and Evaluation

## Making Pickel file from the Model